### ISSUE TITLE
Events & Image Upload

### ISSUE THREAD

**Issue by DrCacho** [suyati/line-control#28]
Hi!

I want to know if it supports events like on Change and how can i connect it with a server side image upload script.

Thanks!

Cacho

**Comment by nmohanan** [comment]
To load text from the textarea to the rich editor you can use:

var $textarea = $("#textarea-id");
$textarea.Editor("setText", $textarea.text());
Then, to bring back the new text to the textarea you can use:

var $textarea = $("#textarea-id");
$textarea.text($textarea.Editor("getText"));
This last one can be done in the onsubmit event, so before the form is submitted, the textarea will be updated.
As of now we are embedding the images into the textarea, for the faster loading time, server side image upload is also possible.
Thanks

**Comment by magraopg** [comment]
Hello I wonder what's wrong, I have a typed text into the textarea, and place an image dragged or placed by the Insert Picture button when I give the post of form, this image no longer appears, only the text appears. If you can help me I am grateful.

thank you

Att: Cesar

**Comment by suneetha15** [comment]
Hi,

We are using LineControl open source rich text editor to one of our application. We  are not able to upload Image from Safari Browser.

Can we able to resolve the issue?. If yes please share the fixes required.

Thanks,
Suneetha

**Comment by TonyFrancis** [comment]
@suneetha15  please tell us which version of safari you are using and line-control.  Can you show the code for the page you are using

**Comment by TonyFrancis** [comment]
@suneetha15 please check after upload image appears in select  box . If it does you need to select the Image to display on text editor otherwise image will not appear in text editor.
If issue is not resolved please reach back to us

**Comment by suneetha15** [comment]
Thanks for the reply.

We are tested with safari 5.1 and 8.0 versions. As i found able to upload image from safari 7.0 and above.

But we are not able to upload from safari 5.1 and IE9.

with that we are also facing few issues while uploading images from IPad.

1. Not able to upload image size >2 mb
2. If we try to upload image with more size (ex: 5 mb), we are facing issue like (problem encountered maximum size exceed the limit) (only in IPad safari

can you please let us know the maximum size image we can upload from editor at one request.

Code snippet:

function LoadEditor() {
            $("#txtEditor").Editor();          
        }

<div>
  <textarea id="txtEditor"></textarea>
</div>

the editor data including image we are copying into asp hidden field and saving into database from codebehind

$("#<%=hdnRichtext.ClientID%>").val($(".Editor-editor").html());


Please help us to resolve its live issue we are facing.

**Comment by suneetha15** [comment]
@TonyFrancis, Yeah from safari 7.0 and above version we are able to select an image. but not able to select image from safari 5.1 version

Thanks !!!

**Comment by TonyFrancis** [comment]
@suneetha15 line-control is made with HTML5 .so browser which only support HTML5.
In this case latest version browser eg IE10 and above

**Comment by suneetha15** [comment]
@TonyFrancis Yes we are disabled image upload from IE9.

can you please let us know the maximum size of image we can upload from editor at one request.

It's Sharepoint application .Net 4.0, we are integrated the editor control for discussion board, where user can upload any size of images.

Thanks

**Comment by TonyFrancis** [comment]
@suneetha15  We have no restriction on image sizes for upload. Please check if there is any restrication in your Sharepoint application

**Comment by suneetha15** [comment]
@TonyFrancis , Yes Sharepoint has limitation to upload max 250 mb at once. we have file attachment also in same page and able to upload files upto 100 mb. 

But with image data its taking more time. 

the editor image data is in base64 format. 

Thanks

**Comment by TonyFrancis** [comment]
@suneetha15  We trying solving for now use this work around 
 <b> Store the image in cloud and include image using URL option </b>
 If this don't work please reach back to us

**Comment by suneetha15** [comment]
Hi @TonyFrancis,

We cannot use URL option, In sharepoint we have to manage in list, because of that we have saving image in base64 format to sharepoint list.

As I found the error (problem encountered when exceed the limit) I'm getting in browser only before request processing to database. 

Thanks

**Comment by xOneca** [comment]
> Yes Sharepoint has limitation to upload max 250 mb at once. we have file attachment also in same page and able to upload files upto 100 mb.

@suneetha15, are you sending the form as `application/x-www-form-urlencoded` or as `multipart/form-data`? Maybe Sharepoint has stricter limit for the former?

**Comment by suneetha15** [comment]
Hi @xOneca 

sorry for delay in response.

Yes we know that sharePoint limitation upto 250 mb. we have restricted also. 

it's web application using c# as code behind language. saving data in base64 format

**Comment by suneetha15** [comment]
Hi

How to move cursor to next line after image inserting in editor.

Thanks in advance

**Comment by m7ammad7assan** [comment]
how to upload the image to server after selecting the image, i dont want to convert it to DATAURI

**Comment by nmohanan** [comment]
You can upload the image with the help of server side scripting, currently we keeps the image as encoded format for the better performance and light weight.
